storage/src/test/java/zipkin2/storage/kafka/streams/TraceStorageTopologyTest.java

```diff
@@ -144,72 +139,67 @@
-    // Then: service name stores are filled
-    KeyValueStore<String, String> serviceNames =
-        testDriver.getKeyValueStore(SERVICE_NAMES_STORE_NAME);
-    List<String> serviceNameList = new ArrayList<>();
-    serviceNames.all().forEachRemaining(serviceName -> serviceNameList.add(serviceName.value));
-    assertThat(serviceNameList).hasSize(2);
-    assertThat(serviceNames.get("svc_a")).isEqualTo("svc_a");
-    assertThat(serviceNames.get("svc_b")).isEqualTo("svc_b");
-    KeyValueStore<String, Set<String>> spanNames =
-        testDriver.getKeyValueStore(SPAN_NAMES_STORE_NAME);
-    assertThat(spanNames.get("svc_a")).containsExactly("op_a");
-    assertThat(spanNames.get("svc_b")).containsExactly("op_b");
-    KeyValueStore<String, Set<String>> autocompleteTags =
-        testDriver.getKeyValueStore(AUTOCOMPLETE_TAGS_STORE_NAME);
-    assertThat(autocompleteTags.get("environment")).containsExactly("dev");
-    // When: clock moves forward
-    Span d = Span.newBuilder()
-        .traceId("d")
-        .id("d")
-        .timestamp(
-            MILLISECONDS.toMicros(traceTtlCheckInterval.toMillis()) + MILLISECONDS.toMicros(20))
-        .build();
-    testDriver.pipeInput(
-        factory.create(spansTopic, d.traceId(), Collections.singletonList(d),
-            traceTtlCheckInterval.plusMillis(1).toMillis()));
-    // Then: Traces store is empty
-    assertThat(traces.get(a.traceId())).isNull();
+    WindowStore<String, List<Span>> traces = testDriver.getWindowStore(TRACES_STORE_NAME);
+    try (final WindowStoreIterator<List<Span>> fetch = traces.fetch(a.traceId(), 0, 10000L)) {
+      assertThat(fetch.hasNext()).isTrue();
+      final KeyValue<Long, List<Span>> next = fetch.next();
+      assertThat(next.value).isEqualTo(spans);
+    }
+    WindowStore<String, Set<String>> spanNames = testDriver.getWindowStore(SPAN_NAMES_STORE_NAME);
+    Map<String, Set<String>> serviceNames = new HashMap<>();
+    try (
+      final KeyValueIterator<Windowed<String>, Set<String>> fetch = spanNames.fetchAll(0, 10000L)) {
+      while (fetch.hasNext()) {
+        final KeyValue<Windowed<String>, Set<String>> next = fetch.next();
+        serviceNames.put(next.key.key(), next.value);
+      }
+    }
+    assertThat(serviceNames).hasSize(2);
+    assertThat(serviceNames.keySet()).contains("svc_a", "svc_b");
+    assertThat(serviceNames.get("svc_a")).containsExactly("op_a");
+    assertThat(serviceNames.get("svc_b")).containsExactly("op_b");
+    WindowStore<String, Set<String>> tagsStore = testDriver.getWindowStore(AUTOCOMPLETE_TAGS_STORE_NAME);
+    Map<String, Set<String>> tags = new HashMap<>();
+    try (
+      final KeyValueIterator<Windowed<String>, Set<String>> fetch = tagsStore.fetchAll(0, 10010L)) {
+      while (fetch.hasNext()) {
+        final KeyValue<Windowed<String>, Set<String>> next = fetch.next();
+        tags.put(next.key.key(), next.value);
+      }
+    }
+    assertThat(tags.get("environment")).containsExactly("dev");
```

Comment:
might be a way in assertj iterator assert to do similar.. something like 
```java
.extracting(next -> next.key.key(), next -> next.value)
.containsExactly(tuple("environment", collections.singleton("dev") 
```

Reply:
give a try. lmk if it looks good